IdentityCore/src/util/NSString+MSIDExtensions.m

```diff
@@ -227,6 +230,11 @@ + (NSString *)msidWWWFormURLEncodedStringFromDictionary:(NSDictionary *)dict
          {
              [encodedString appendFormat:@"=%@", encodedValue];
          }
+        
+         else if ((encodedValue && ![encodedValue isKindOfClass:[NSNull class]]) && !encodedValue.length)
```

Comment:
if encodedValue is nil, this logic will not attach it to encodedString. Is that okay? Making encodedValue = @"" if it is nilOrBlank might be useful?

Reply:
Yeah good point. I think we should append "=" even for nil or blank scenario. I believe that calling APIs normally convert URLs first into a dictionary with multiple key-value pairs from &key1=value1 substrs and then add some additional key-value pairs and then convert it back to a new URL. I assumed that the check for NULL or blank would be redundant, as that might never happen was my understanding.

I think we should either 
1) create an error for nilorblank, as that might indicate an error in the calling API's logic that converted the original URL into a query dictionary
2) Simply add "=" to the keyword regardless. 

Thanks for your suggestion.